python_modules/libraries/dagster-dbt/dagster_dbt/components/dbt_project/component.py

```diff
@@ -318,17 +329,33 @@ def asset_key_for_model(self, model_name: str) -> AssetKey:
         ).key
 
 
-class ProxyDagsterDbtTranslator(DagsterDbtTranslator):
-    # get_description conflicts on Component, so cant make it directly a translator
-
-    def __init__(self, fn: TranslationFn, settings: Optional[DagsterDbtTranslatorSettings]):
-        self._fn = fn
+class DagsterDbtComponentTranslator(DagsterDbtTranslator):
+    def __init__(
+        self, component: DbtProjectComponent, settings: Optional[DagsterDbtTranslatorSettings]
+    ):
+        self._component = component
         super().__init__(settings)
 
-    def get_asset_spec(self, manifest, unique_id, project):
-        base_asset_spec = super().get_asset_spec(manifest, unique_id, project)
-        dbt_props = get_node(manifest, unique_id)
-        return self._fn(base_asset_spec, dbt_props)
+    def get_asset_spec(
+        self, manifest: Mapping[str, Any], unique_id: str, project: Optional[DbtProject]
+    ) -> dg.AssetSpec:
+        # if the component overrides the default implementation of get_asset_spec, use that
```

Comment:
this comment reads weird to me since we are always going through the component to get the base asset spec 

what i see is closer to `# get the original spec from the component class (potentially overriden) then apply translation if provided`